addons/web_editor/controllers/main.py

```diff
@@ -270,130 +270,182 @@ def get_image_info(self, image_id=None, xml_id=None):
-            ]
-
-            # Fetch the arch of each kept file, in each bundle
-            for bundle_data in scss_files_data_by_bundle:
-                for i in range(0, len(bundle_data[1])):
-                    url = bundle_data[1][i]
-                    url_info = url_infos[url]
-
-                    content = None
-                    if url_info["customized"]:
-                        # If the file is already customized, the content is found in the corresponding attachment
-                        content = base64.b64decode(custom_attachments.filtered(lambda a: a.url == url).datas)
-                    else:
-                        # If the file is not yet customized, the content is found by reading the local scss file
-                        module = url_info["module"]
-                        module_path = get_module_path(module)
-                        module_resource_path = get_resource_path(module, url_info["resource_path"])
-                        if module_path and module_resource_path:
-                            module_path = os.path.join(os.path.normpath(module_path), '') # join ensures the path ends with '/'
-                            module_resource_path = os.path.normpath(module_resource_path)
-                            if module_resource_path.startswith(module_path):
-                                with open(module_resource_path, "rb") as f:
-                                    content = f.read()
-
-                    bundle_data[1][i] = dict(
-                        url = "/%s/%s" % (url_info["module"], url_info["resource_path"]),
-                        arch = content,
-                        customized = url_info["customized"],
-                    )
+            scss_files_data_by_bundle = self.get_resources('scss', views, only_user_custom_files, bundles_restriction)
 
         return dict(
-            views = get_views and views or [],
-            scss = get_scss and scss_files_data_by_bundle or [],
+            views=get_views and views or [],
+            scss=get_scss and scss_files_data_by_bundle or [],
+            js=get_js and js_files_data_by_bundle or [],
         )
 
-    def save_scss_view_hook(self):
+    def get_resources(self, file_type, views, only_user_custom_files, bundles_restriction):
+        resources = self.load_resources(views, only_user_custom_files, file_type, bundles_restriction)
+        url_infos = resources['url_infos']
+        custom_attachments = resources['custom_attachments']
+        files_data_by_bundle = resources['files_data']
+        data = {
+            'files_data': files_data_by_bundle,
+            'bundles_restriction': bundles_restriction,
+            'url_infos': url_infos,
+            'custom_attachments': custom_attachments,
+        }
+        self.filter_bundles_files_and_fetch(data)
+        return files_data_by_bundle
+
+    def load_resources(self, views, only_user_custom_files, resources_type, bundles_restriction):
```

Comment:
Same: use the same order of parameters everywhere. But in fact, no need of a load_resources function since it is only called one time by get_resources. I would only define one function: `load_resources` which does both the job of `get_resources` and `load_resources`. This will also make the diff easier to read.